benefits/oauth/views.py

```diff
@@ -0,0 +1,44 @@
+from django.shortcuts import redirect
+from django.urls import reverse
+from django.utils.decorators import decorator_from_middleware
+
+from authlib.integrations.django_client import OAuth
+
+from benefits.core import middleware, session
+
+
+_oauth = OAuth()
+
+
+@decorator_from_middleware(middleware.AgencySessionRequired)
+@decorator_from_middleware(middleware.VerifierSessionRequired)
+def login(request):
+    verifier = session.verifier(request)
+    if not verifier.requires_authentication:
+        return redirect(session.origin(request))
```

Comment:
While this is presumably working now, I worry a bit about the complexity of our session states, and having bugs that put users in incompatible states. Will noodle on that a bit as we go.

As an example of how to simplify, do we actually need to check `AgencySessionRequired`, `VerifierSessionRequired`, and `verifier.requires_authentication` here? Ditto for `authorize`.

Reply:
We definitely need to check `VerifierSessionRequired` since we need the `verifier` later to get the `oauth_client`.

`AgencySessionRequired` isn't technically needed / since `verifier` comes from the agency originally. But it's a valid sanity check - the user shouldn't be here if they don't have an agency in their session.

On `requires_authentication`... I could go either way. Theoretically the user doesn't get here unless their verifier requires auth, so we could assume that.

Reply:
I posted [a follow-up](https://github.com/cal-itp/benefits/pull/414#discussion_r847845386) to this and other comments in another thread - refactored the code to address simplification suggestion